rpcclient/chain.go

```diff
@@ -900,11 +900,7 @@ func (r FutureGetCFilterHeaderResult) Receive() (*wire.MsgCFHeaders, error) {
 	}
 
 	// Assign the hash to a headers message and return it.
-	var msgCFHeaders wire.MsgCFHeaders
-	err = msgCFHeaders.AddCFHeader(headerHash)
-	if err != nil {
-		return nil, err
-	}
+	msgCFHeaders := wire.MsgCFHeaders{PrevFilterHeader: *headerHash}
```

Comment:
Should this be more like:

```
	msgCFHeaders := wire.MsgCFHeaders{
		FilterHashes: []*chainhash.Hash{headerHash}}
```?

Reply:
Neither is ideal, though I preferred putting the result in PrevFilterHeader, because at least the field is supposed to be a header, not a hash. The fact that they happen to be of type `*chainhash.Hash` is kind of a coincidence in a sense, but I think of them as distinct types.

Reply:
I think I get your reasoning now. I wonder if it'd be better to either:

* Make a new type for the RPC response instead of reusing the wire type, and return the real `PrevFilterHeader`, the current `FilterHash`, and the current `FilterHeader` from the RPC server, or
* Just return the `*chainhash.Hash` without stuffing it into a wire format that no longer works the same way from the RPC client?

Reply:
Yeah, that's fair. Probably makes more sense for this to return a `*chainhash.Hash`. By the same token though, I think `GetCFilter` should return a `*gcs.Filter` instead of a `*wire.MsgCFilter`.

Reply:
That also makes sense to me.

Reply:
We can make those changes later; I've made a `neutrino` PR that works as is. Everything seems to work fine.